.github/workflows/test.yml

```diff
@@ -0,0 +1,37 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10", "3.11"]
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Build data-validation
+      id: build-data-validation
+      uses: ./.github/reusable-build
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install test dependencies
+      run: |
+        pip install pytest scikit-learn scipy
```

Comment:
Can the test dependencies get moved to `setup.py` or `pyproject.toml`? It's easier to keep track of dependencies if they're all in one place.

Reply:
These can be added to `pyproject.toml`, but can't be moved there since we can't install (AFAIK) just the optional dependencies for testing (See https://github.com/pypa/pip/issues/11440#issuecomment-1465817054) as rest of the project is installed by the artifact of built wheel inside docker.

Reply:
Also, the side effects of adding it in `pyproject.toml`: https://github.com/tensorflow/data-validation/actions/runs/11211638538/job/31160932203?pr=260